py3status/modules/whatismyip.py

```diff
@@ -49,8 +52,24 @@ class Py3status:
     mode = 'ip'
     negative_cache_timeout = 2
     timeout = 5
-    url = 'http://ultrabug.fr/py3status/whatismyip'
-    url_geo = 'http://ip-api.com/json'
+    url_geo = URL_GEO_NEW_DEFAULT
+
+    class Meta:
+        deprecated = {
+            'remove': [
+                {
+                    'param': 'url',
+                    'msg': 'obsolete parameter, use `url_geo` instead',
+                },
+            ],
+        }
+
+    def post_config_hook(self):
+        # Backwards compatibility
+        if self.url_geo == URL_GEO_NEW_DEFAULT:
+            self.format = self.format.replace('{country}', '{country_name}')
+        elif self.url_geo == URL_GEO_OLD_DEFAULT:
+            self.format = self.format.replace('{ip}', '{query}')
```

Comment:
although this is a nice approach it is flawed eg `'{country:^20}'` is a valid format string.

I think it might just be easier to do something like

```
def post_config_hook(self):
    if self.url_geo == URL_GEO_NEW_DEFAULT:
        self.substitutions = [('country_name', 'country')]

def _get_my_ip_info(self):
    ...
    for field, sub in self.substitutions:
        info[sub] = info.get(field)
```

I'm sure this can be improved